core/src/main/java/org/apache/cxf/common/spi/ClassGeneratorClassLoader.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.cxf.common.spi;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.cxf.Bus;
+import org.apache.cxf.BusFactory;
+
+public class ClassGeneratorClassLoader {
+    //TODO handle that with system property
+    private static final boolean DEBUG = false;
```

Comment:
I think we could delete this debug flag and related code. It could be easily achieved by introducing respective extension fe `DumpingClassGeneratorClassLoader extends ClassGeneratorClassLoader`

Reply:
ok. I was thinking it is a cool feature to add. if I add a system property for that but ok. I will remove it.

Reply:
Thinking out loud: if you replace the boolean by a string "GENERATED_CLASS_OUTPUT", wouldnt it help the build time generation? (null=deactivate=default)

Reply:
and make it public... @reta  ok for you ?

Reply:
or System.getProperty ;)

Reply:
If you really need to have this capability (vs providing specialized extension), I would go with system property with location to dump generated bits.

Reply:
I have removed it. I do not need it. It was during debug and think it was a cool feature for users to debug their apps but ok.

Reply:
👍 thanks!